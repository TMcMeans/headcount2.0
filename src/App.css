@import url(https://fonts.googleapis.com/css?family=Lora|Roboto:400,500);

nav {
  width: 100%;
  background: lightblue;
}

h1 {
  text-align: center;
  font-family: 'Lora', sans-serif;
  font-size: 40px;
  margin-top: 0;
  opacity: 0.8;
}


