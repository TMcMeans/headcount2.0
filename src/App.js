import React, { Component } from 'react';
import kinderData from './data/kindergartners_in_full_day_program.js';
import DistrictRepository from './helper.js';

import './App.css';
import CardContainer from './CardContainer.js'
import SearchForm from './SearchForm.js'
import CompareCard from './CompareCard.js'

class App extends Component {
  constructor() {
    super() 
      this.state = {
        districts: [],
        schoolName: '',
        data: new DistrictRepository(kinderData),
        compareSelections: []
    }
  }

  componentDidMount() {
    this.populateAllCards(kinderData)
  }

 populateAllCards = (data) => {
  const repository = new DistrictRepository(data)
  this.setState({
    districts: repository.findAllMatches()
  })
 }

 filterCards = (school) => {
  this.setState({
    schoolName: school
  })
 }

 addCompareSelections = (e) => {

  const card = e.target.closest('.card')
  const name = card.children[0].innerText

  const schoolObj = {...this.state.data.findAllMatches(name)} 

  this.setState({
    compareSelections: [{...schoolObj[0]}, ...this.state.compareSelections] 
  })
 }

 render() {
    const { districts, data, schoolName, compareSelections } = this.state
    if (districts.length > 0) { 
      return (
        <div>
          <h1>Headcount 2.0</h1>
          <CompareCard compareSelections={compareSelections} />
          <SearchForm filterCards={this.filterCards} />
          <CardContainer districts={data.findAllMatches(schoolName)} 
                        addCompareSelections={this.addCompareSelections}
          />
        </div>
      )
    } else {
      return (
        <div>
          <h1>Headcount 2.0</h1>
        </div>
      )
    }
  }
}

export default App;
