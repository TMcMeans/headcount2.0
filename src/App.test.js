import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import App from './App';

describe('App', () => {
  it.skip('should display school districts', () => {
    // Setup && Execution
    const wrapper = shallow(<App />)

    // Expectation
    expect(wrapper).toMatchSnapshot()
  })

})