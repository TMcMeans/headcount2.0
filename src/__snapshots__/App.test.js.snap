// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should display school districts 1`] = `
<div>
  <h1 />
</div>
`;
